docs/core_concepts/11_persistent_storage/index.md

```diff
@@ -0,0 +1,283 @@
+# Persistent Storage
+
+Persistent storage refers to any method of storing data that remains intact and accessible even after a system is powered off, restarted, or experiences a crash.
+
+In the context of Windmill, the stakes are: **where to store the data manipulated by Windmill** (ETL, data ingestion and preprocessing, data migration and sync etc.)?
+
+:::info TLDR
+
+Within Windmill, it is recommended to only store Windmill-specific elements ([resources](../3_resources_and_types/index.md), [variables](../2_variables_and_secrets/index.md) etc.). For data, we suggest using storage service providers to be targeted from Windmill.
+
+<br/>
+
+This present document gives a list of trusted services to use alongside Windmill.
+
+:::
+
+
+## You already have your own database
+
+Then we have nothing to say.
+
+If your service provider is already part of [our list of integrations](../../integrations/0_integrations_on_windmill.md), just add your database as a [resource](../../core_concepts/3_resources_and_types/index.md).
+
+Otherwise, create access to your service provider through a [new resource type](../../core_concepts/3_resources_and_types/index.md#create-a-resource-type) (and if you want, [share the schema on our Hub](../../misc/1_share_on_hub/index.md)).
+
+## Within Windmill: not recommended
+
+Windmill is not designed to store data that goes beyond the execution of a script or flow. Indeed, for each computation the worker executing is not the same as the previous computation, so the data would have to be retrieved from another location.
```

Comment:
to store heavy data. However, one can store a transient state that can be represented as small json (see setState [with link and stuff to reference])

Reply:
you can actually say it afterwards, so maybe just a bit of rewording